operator-framework-core/src/main/java/io/javaoperatorsdk/operator/processing/dependent/workflow/Condition.java

```diff
@@ -1,10 +1,11 @@
 package io.javaoperatorsdk.operator.processing.dependent.workflow;
 
+import java.util.Optional;
+
 import io.fabric8.kubernetes.api.model.HasMetadata;
 import io.javaoperatorsdk.operator.api.reconciler.Context;
-import io.javaoperatorsdk.operator.api.reconciler.dependent.DependentResource;
 
 public interface Condition<R, P extends HasMetadata> {
 
-  boolean isMet(DependentResource<R, P> dependentResource, P primary, Context<P> context);
+  boolean isMet(P primary, Optional<R> secondary, Context<P> context);
```

Comment:
I'm not sure about this change, various dependent resource implementations can have some additional information. Like dependent resources for external resources with state, information about the state.

Reply:
Also this is a non-backwards compatible change. But that not on issue for now IMHO :)

Reply:
Paramters should not be Optional, maybe just document it that it can be null.

Reply:
DependentResource classes should not be accessible directly to user-controlled code, imo. While unlikely, someone could call `reconcile` on it from the condition, for example… State should be conveyed via the `Context`, imo.

Reply:
Agree with the context part.  But specifically speking about not that part, although don't see it clearly yet. So let's park that for now. 
If you think about it will be more complex than that where there an additional state in a config map for example for an external dependent resource 
But fine with this now.

Reply:
Except the Optional part, that should not be there according Java conventions.